### PR TITLE
Online training

### PR DESCRIPTION
**Pull request by Chris-Pedersen**
The major 2 changes to the code required for online training are:
1. A flag on all relevant tensors to track gradients (`requires_grad=True`)
2. Removing all inplace operations, which break the backprop.

Have done this for the pseudospectral kernel, not yet Arakawa.

### CODE REVIEW

**Comment by Chris-Pedersen** [comment]
Just gonna merge this for now and get back to other kernels later

**Comment by Chris-Pedersen** [comment]
Online training was _working_ for the pseudospectral kernel in that the compute would run without crashes.. but the loss was not dropping. A few potential reasons for this; one is that the kernel itself is in physical units, while our networks are of course normalised. So at every timestep we have to transform in and out of physical units, crossing ~8 orders of magnitude. We know the determinant in the inversion is already very small, and the gradients would need to flow through this as well, so I suspect the issue is there. But I'm not rewriting the full kernel right now, so we're going to sideline this for the time being.
